File: FestivalJava/Services/src/main/java/ModelServices/Interfaces/IServiceUser.java
package ModelServices.Interfaces;

import Domain.User;
import Validation.Exceptions.ValidatorException;

import java.util.List;

/**
 * Created by Sebi on 28-Mar-17.
 */
public interface IServiceUser {
    /*
    Saves an user
     */
    void save(User user) throws ValidatorException;

    /*
    Deletes an user
     */
    void delete(String username);

    /*
    Updates an user
     */
    void update(String id, User user) throws ValidatorException;

    /*
    Gets a user by id
     */
    User getById(String username);

    /*
    Gets all users
     */
    List<User> getAll();

    /*
    Get user by name and password
     */
    User getByNamePass(String username, String password);
}

File: FestivalC#/Festival/Services/Service/ModelServices/IServiceUser.cs
using Festival.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Service
{
    public interface IServiceUser
    {
        /*
         * Saves an user
         */
        void save(User user);

        /*
         * Deletes an user
         */
        void delete(string username);

        /*
         * Updates an user
         */
        void update(string username, User user);

        /*
         * Gets an user
         */
        User getById(string username);

        /*
         * Gets all users
         */
        List<User> getAll();

        /*
         * Gets an user by name + pass
         */
        User getByNamePass(string username, string password);
    }
}
